packages/platform-core/src/telemetry/translate_error_to_error_details.ts

```diff
@@ -0,0 +1,40 @@
+import { SerializableError } from './serializable_error';
+import { TelemetryPayload } from './telemetry_payload';
+
+/**
+ * Translates an error into the shape of error details for telemetry
+ */
+export const translateErrorToErrorDetails = (
+  error?: Error,
+): TelemetryPayload['error'] => {
+  if (!error) {
+    return undefined;
+  }
+  let currentError: Error | undefined = error;
+  let errorDetails: TelemetryPayload['error'];
+
+  try {
+    while (currentError) {
+      const serializedError = new SerializableError(currentError);
+      const errorDetail: TelemetryPayload['error'] = {
+        name: serializedError.name,
+        message: serializedError.message,
+        stack: serializedError.stack ?? '',
+      };
+
+      if (errorDetails) {
+        // this reverses the nesting so lowest level error is the top level error in our telemetry
+        errorDetail.cause = errorDetails;
+      }
```

Comment:
If we do the reversal. the `cause` field doesn't make sense. May be
```suggestion
      if (errorDetails) {
        // this reverses the nesting so lowest level error is the top level error in our telemetry
        errorDetail.caused = errorDetails;
      }
```